share/patch/patch_db_zonemaster_backend_ver_11.0.3.pl

```diff
@@ -0,0 +1,163 @@
+use strict;
+use warnings;
+
+use JSON::PP;
+use Try::Tiny;
+
+use Zonemaster::Backend::Config;
+
+my $config = Zonemaster::Backend::Config->load_config();
+
+my %patch = (
+    mysql       => \&patch_db_mysql,
+    postgresql  => \&patch_db_postgresql,
+    sqlite      => \&patch_db_sqlite,
+);
+
+my $db_engine = $config->DB_engine;
+print "engine: $db_engine\n";
+
+if ( $db_engine =~ /^(MySQL|PostgreSQL|SQLite)$/ ) {
+    $patch{ lc $db_engine }();
+}
+else {
+    die "Unknown database engine configured: $db_engine\n";
+}
+
+sub _update_data {
+    my ( $dbh ) = @_;
+
+    my $json = JSON::PP->new->allow_blessed->convert_blessed->canonical;
+
+    my ( $row_total ) = $dbh->selectrow_array( 'SELECT count(*) FROM test_results' );
+    print "count: $row_total\n";
+
+    # depending on the resources available to select all data in database
+    # update $row_count to your needs
+    my $row_count = 50000;
+    my $row_done = 0;
+    while ( $row_done < $row_total ) {
+        print "row_done/row_total: $row_done / $row_total\n";
+        my $sth1 = $dbh->prepare( 'SELECT hash_id, results FROM test_results ORDER BY id ASC LIMIT ?,?' );
+        $sth1->execute( $row_done, $row_count );
+        while ( my $row = $sth1->fetchrow_arrayref ) {
+            my ( $hash_id, $results ) = @$row;
+
+            next unless $results;
+
+            my @records;
+            my $entries = $json->decode( $results );
+
+            foreach my $m ( @$entries ) {
+                my $r = [
+                    $hash_id,
+                    $m->{level},
```

Comment:
Testing the patch script I get a foreign key constraint error, I guess this should be `$levels{$m->{level}}` with `levels` defined as `%levels = Zonemaster::Engine::Logger::Entry->levels()`.

Reply:
Good point. I forgot to update the script when changing the value stored. I'll provide an update.

Reply:
Fixed